network/src/sync/blocks.rs

```diff
@@ -89,10 +106,11 @@ impl<S: Storage + Send + std::marker::Sync + 'static> Node<S> {
         };
 
         info!(
-            "Received block from {} of epoch {} with hash {:?}",
+            "Received block from {} of epoch {} with hash {:?} (current head {})",
             remote_address,
             block_struct.header.time,
-            hex::encode(block_struct.header.get_hash().0)
+            hex::encode(block_struct.header.get_hash().0),
```

Comment:
```suggestion
            block_struct.header.get_hash()
```
it's not something from this PR, but this alternative, simpler form should suffice, paired with `{}` instead of `{:?}` in the format literal